src/alloc/isolated_alloc.rs

```diff
@@ -39,38 +42,57 @@ impl IsolatedAlloc {
 
     /// Expands the available memory pool by adding one page.
     fn add_page(&mut self, page_size: usize) -> (*mut u8, &mut DenseBitSet<usize>) {
-        assert_ne!(page_size, 0);
-
         let page_layout = unsafe { Layout::from_size_align_unchecked(page_size, page_size) };
         let page_ptr = unsafe { alloc::alloc(page_layout) };
         // `page_infos` has to have one-eighth as many bits as a page has bytes
         // (or one-64th as many bytes)
-        self.page_infos.push(DenseBitSet::new_empty(page_size / 8));
+        self.page_infos.push(DenseBitSet::new_empty(page_size / COMPRESSION_FACTOR));
         self.page_ptrs.push(page_ptr);
         (page_ptr, self.page_infos.last_mut().unwrap())
     }
 
-    /// For simplicity, we allocate in multiples of 8 bytes with at least that
-    /// alignment.
+    /// For simplicity, we serve allocations in multiples of COMPRESSION_FACTOR
+    /// bytes with at least that alignment.
     #[inline]
     fn normalized_layout(layout: Layout) -> (usize, usize) {
-        let align = if layout.align() < 8 { 8 } else { layout.align() };
-        let size = layout.size().next_multiple_of(8);
+        let align =
+            if layout.align() < COMPRESSION_FACTOR { COMPRESSION_FACTOR } else { layout.align() };
+        let size = layout.size().next_multiple_of(COMPRESSION_FACTOR);
         (size, align)
     }
 
+    /// If the allocation is greater than a page, then round to the nearest page #.
+    /// Since we pass this into the global allocator, it's more useful to return
+    /// a `Layout` instead of a pair of usizes.
+    #[inline]
+    fn huge_normalized_layout(layout: Layout, page_size: usize) -> Layout {
+        // Allocate in page-sized chunks
+        let size = layout.size().next_multiple_of(page_size);
+        // Align probably *shouldn't* ever be greater than the page size,
+        // but just in case
```

Comment:
It's completely legal to ask for a 16k align on a 4k page size system.

Reply:
Fair enough, lol